.github/workflows/build-and-test.yml

```diff
@@ -264,6 +264,7 @@ jobs:
         with:
           go-version: ${{ matrix.go-version }}
           cache: false
```

Comment:
BTW if anyone (like me) reads this and wonders "why is it caching 1.23.6 if `cache: false` is set?", then see https://github.com/actions/setup-go/issues/549. That config only affects caching of modules, not the toolchain.